x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/action_connector_form/helpers.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { cloneDeep } from 'lodash/fp';
+
+const ELASTICSEARCH_PROVIDER = 'elasticsearch';
+const MIN_ALLOCATIONS = 0;
+const DEFAULT_NUM_THREADS = 1;
+
+export const getInferenceApiParams = (data: any, enforceAdaptiveAllocations: boolean) => {
```

Comment:
Note on this function - ideally we should be doing this in the form serializer but because right now we need `enforceAdaptiveAllocations` (which isn't available outside of the component) we need to keep this as an external function. 
Once this change is included in all environments and we no longer need that flag - this will be moved to the serializer. cc @jcger 🙏